File: in_Java/0028 Implement strStr().java
class Solution {
    public int strStr(String haystack, String needle) {
        if(needle == null || needle.length() == 0) return 0;
        if(haystack == null || haystack.length() == 0) return -1;
        int n = needle.length();
        for(int i = 0; i <= haystack.length() - n; i++){
            if(haystack.substring(i,i+n).equals(needle)) return i;
        }
        return -1;
    }
}

File: in_Python/0028 Implement strStr().py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        if len(needle) == 0:
            return 0
        l = len(needle)
        for i in range(len(haystack) - l + 1):
            s = haystack[i : i + l]
            if s == needle:
                return i
        return -1